lib/rmt/mirror.rb

```diff
@@ -1,205 +1,38 @@
-require 'rmt/downloader'
-require 'rmt/gpg'
-require 'repomd_parser'
-require 'time'
-
 class RMT::Mirror
-  class RMT::Mirror::Exception < RuntimeError
-  end
+  RPM_FILE_NEEDLE = 'repodata/repomd.xml'.freeze
+  DEB_FILE_NEEDLE = 'Release'.freeze
 
-  include RMT::Deduplicator
-  include RMT::FileValidator
+  attr_reader :logger, :base_dir, :mirror_sources, :is_airgapped, :repository
 
-  def initialize(mirroring_base_dir: RMT::DEFAULT_MIRROR_DIR, logger:, mirror_src: false, airgap_mode: false)
-    @mirroring_base_dir = mirroring_base_dir
+  def initialize(repository:, base_dir:, logger:, mirror_sources: false, is_airgapped: false)
+    @repository = repository
     @logger = logger
-    @mirror_src = mirror_src
-    @airgap_mode = airgap_mode
-    @deep_verify = false
-
-    # don't save files for deduplication when in offline mode
-    @downloader = RMT::Downloader.new(logger: logger, track_files: !airgap_mode)
+    @base_dir = base_dir
+    @mirror_sources = mirror_sources
+    @is_airgapped = is_airgapped
   end
 
-  def mirror_suma_product_tree(repository_url:)
-    # we have an inconsistency in how we mirror in offline mode
-    # in normal mode we mirror in the following way:
-    # base_dir/repo/...
-    # however, in offline mode we mirror in the following way
-    # base_dir/...
-    # we need this extra step to ensure that we write to the public directory
-    base_dir = mirroring_base_dir
-    base_dir = File.expand_path(File.join(mirroring_base_dir, '/../')) if mirroring_base_dir == RMT::DEFAULT_MIRROR_DIR
-
-    repository_dir = File.join(base_dir, '/suma/')
-    mirroring_paths = {
-      base_url: URI.join(repository_url),
-      base_dir: repository_dir,
-      cache_dir: repository_dir
+  def detect_repository_type
```

Comment:
Imho the 'detect_' prefix is superfluous.

Reply:
I agree, we keep it in mind and change the naming when we switch to our new implementations!